### DISCUSSION TITLE
Example Change #2: IATI Identifier Rules

### DISCUSSION THREAD

**Discussion by robredpath** [IATI/standard-process#5] [Demos and examples]
In our meeting earlier this month, we looked an issue that exists with IATI Identifiers starting or ending with newlines that made some activities impossible to search for. This is an example of the kind of real-world issues that we deal with in standard governance.

We've copied the example into this repo as an example change request: https://github.com/IATI/standard-process/blob/main/example_changes/iati_identifier_rules.md

It would be useful to hear views from the group on what IATI's response to this issue should be. We can set aside for a moment the existing upgrade process and consider what we think a good response looks like. As we discuss, we may uncover principles that can contribute to the https://github.com/IATI/standard-process/discussions/2 that @stevieflow has opened.

**Comment by bill-anderson** [comment]
1. Agree it should be 'must' and validator should record error
2. Agree this should have no impact on the data store
3. BUT I think this also speaks to the argument for an augmented version of the datastore where this error is cleaned. (So long as the augmented version documentation is very specific about all instances of cleaning.

**Comment by davidmegginson** [comment]
We can handle this by publishing guidance on normalising identifiers for comparison or database ingestion:

- all leading and trailing characters with the Unicode property White_Space=yes shall be stripped
- all sequences of internal characters with the Unicode property White_Space=yes shall be replaced with a single space character U+0020
- all characters of Unicode category "Upper Case Letter" shall be replaced with the corresponding Unicode "Lower Case Letter"

I expect we'll find a few pathological cases where "identifier" and "  iDenTifier  " refer to two different activities, but I think they'll be few enough that we can work with data providers 1:1 to resolve them
